### DISCUSSION TITLE
Zooming to a certain point instead of mouse position

### DISCUSSION THREAD

**Discussion by blairmcalpine** [d3/d3#3643] [Q&A]
I'm using d3@7.3.0, and I would like to make a zoomable/pannable timeline with d3 zoom and a d3 timescale. My issue is that I want the timeline to zoom in and out of the center of the timeline as opposed to where my mouse position is. Here is my current behaviour: 
![chrome-capture-2023-2-15](https://user-images.githubusercontent.com/97198784/225365816-d299afea-3b24-4a4b-a609-056aec4c1cee.gif)

And an example behaviour that I would like it to follow (note the zooming being on the center point of the timeline):
![AcsClient_XlkGMni9iu](https://user-images.githubusercontent.com/97198784/225367780-57f6a287-5912-4068-9fe0-2511dc448fbf.gif)

I've noticed that older versions of d3 had the zoom.center() function that seemed to cater to this behaviour, ([an example that used it in a similar way to what I want](https://observablehq.com/d/48ff4a64c477cc79)) but it doesn't exist anymore. What would be the best way to implement this? Here is a general layout of my current code for reference as well (very simple of course):
```
const scale = d3
        .scaleTime()
        .domain([thirtyMinsAgo, now])
        .range([0, timelineWidth - 1])

const zoom = d3
        .zoom()
        .scaleExtent([0.0025, 250])
        .on('zoom', ({ transform }) => {
          scaleRef.current = transform.rescaleX(scale);
          scaleTimeline(scaleRef.current);
        })

svg.call(zoom);
```

**Comment by Fil** [comment]
Ref. https://github.com/d3/d3-zoom/pull/212
